### ISSUE TITLE
[Feature Request][Keras] Allow loss function using multiple tensors as input

### ISSUE THREAD

**Issue by howl-anderson** [tensorflow/tensorflow#37818]
Some loss functions not only depend on `y_pred` and `y_true` but also depend on other tensors, such as a CTC loss (related to 
* https://github.com/ysoullard/CTCModel/blob/d9e87fb24a6f7446457de5c89004404a900ce00b/CTCModel.py#L108
* https://github.com/robmsmt/KerasDeepSpeech/blob/553638821c996cb6049e34185babacdb8f00c215/model.py#L145
* https://github.com/ypwhs/captcha_break/blob/master/ctc_2019.ipynb

), 

Yolo3 loss (related to 
* https://github.com/qqwweee/keras-yolo3/blob/e6598d13c703029b2686bc2eb8d5c09badf42992/train.py#L131
* https://github.com/experiencor/keras-yolo3/blob/768c524f277adbfd26c2f44d73cb1826bbaf2d10/yolo.py#L361

)

 and CRF loss function (relate to 

* tensorflow/addons#1363
* tensorflow/addons#377

). 

Maybe we can provide a container named `GroupedTensorContainer` to make it works like below:

```python
x_np, y_np = get_test_data()

x_input = tf.keras.layers.Input(shape=x_np.shape[1:])

# NOTE: tensors can from different layers
pred_y, loss_required_1, loss_required_2 = SomeLayer(5)(x_input)

canned_tensors = GroupedTensorContainer(visiable=pred_y, hidden=(loss_required_1, loss_required_2))

model = tf.keras.Model(x_input, canned_tensors)

model.compile("adam", loss=SomeLoss())
model.fit(x_np, y_np)
model.evaluate(x_np, y_np)

container = model.predict(x_np)
container  # this is the visible tensor
```

while the `SomeLoss` has such a prototype:
```python
class SomeLoss:
    def __call__(self, y_true, y_pred, sample_weight=None):
        pred_y = y_pred.visiable_tensor
        loss_required_1, loss_required_2 = y_pred.hidden_tensors

       loss = compute_loss(
            y_true, pred_y, loss_required_1, loss_required_2
        )

        return loss
```

**System information**
- TensorFlow version (you are using): 2.1
- Are you willing to contribute it (Yes/No): Yes

**Comment by howl-anderson** [comment]
@pavithrasv Please take a look, thank you!

**Comment by howl-anderson** [comment]
@pavithrasv Could you please take a look?
Or anyone from the official team willing to take a look? This issue already waiting for more than one year, but I am still want to figure out: the idea is good or not.

**Comment by chunduriv** [comment]
@howl-anderson, Sorry for the late response. Is this still an issue for you?

Can you please try recent TFv2.7 and let us know whether it is persisting. Also, share a simple standalone code to reproduce the issue. Thanks!
